From 177fda86451cdeaaea8ed409e6d711b670699a97 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ga=C3=ABl=20Bonithon?= <gael@xfce.org>
Date: Tue, 6 Feb 2024 18:14:04 +0100
Subject: [PATCH] screen: Fix wrong assert

It's always been wrong (or has been for a long time) but de3e7aac
revealed it, because now it's no longer disabled by building with
--disable-debug.

Fixes: de3e7aac72fdcd3e62d69f37ec2570e5d668950a
Closes: #299
---
 terminal/terminal-screen.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/terminal/terminal-screen.c b/terminal/terminal-screen.c
index 6e48b522..dc931ec7 100644
--- a/terminal/terminal-screen.c
+++ b/terminal/terminal-screen.c
@@ -1892,7 +1892,7 @@ terminal_screen_paste_unsafe_text (TerminalScreen *screen,
 {
   GtkWidget *dialog;
 
-  g_return_if_fail (original_clipboard != GDK_SELECTION_CLIPBOARD && original_clipboard != GDK_SELECTION_PRIMARY);
+  g_return_if_fail (original_clipboard == GDK_SELECTION_CLIPBOARD || original_clipboard == GDK_SELECTION_PRIMARY);
 
   dialog = terminal_screen_unsafe_paste_dialog_new (screen, text);
   gtk_widget_show_all (dialog);
-- 
GitLab


